### PR TITLE
[bzip2] add a "tool" feature to control the build of bzip2.exe

### LINKED ISSUE #26997: [bzip2] bzip2 should not build bzip2.exe unless requested
**Is your feature request related to a problem? Please describe.**
It is not.

**Proposed solution**
bzip2 needs a `tool` feature like curl's

**Describe alternatives you've considered**

**Additional context**
(I am attempting to consume bzip2 in the Windows source tree, and it fails to build bzip2.exe for x86. Regardless of this, however, bzip2 should not emit an EXE unless it is requested.)

### PR DESCRIPTION
**Pull request by DHowett**
This is modelled after curl[tool], but it is on by default for compatibility with the existing portfile.

Fixes #26997

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

### CODE REVIEW

**Comment by DHowett** [comment]
The bot's comments are for an earlier commit. I cannot resolve the license issue. I do not feel confident in fixing the deprecated cmake file.

**Comment by DHowett** [comment]
Oh, interesting. Thanks! What happened here?

File: ports/bzip2/portfile.cmake [hunk 1]
```diff
@@ -11,11 +11,18 @@ vcpkg_extract_source_archive_ex(
     PATCHES fix-import-export-macros.patch
 )
 
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    INVERTED_FEATURES
```
Comment (JonLiu1993) [inline ports/bzip2/portfile.cmake:15]: @DHowett ,Thanks for your pr, could you please clarify why this feature added by `INVERTED_FEATURES`?

**Comment by JonLiu1993** [comment]
@DHowett ,Thanks for your pr, please consider migrating them to the new functions:
````
vcpkg_install_cmake -> vcpkg_cmake_install (from port vcpkg-cmake)
vcpkg_configure_cmake -> vcpkg_cmake_configure (Please remove the option PREFER_NINJA) (from port vcpkg-cmake)
vcpkg_fixup_cmake_targets -> vcpkg_cmake_config_fixup (from port vcpkg-cmake-config)
````
In the ports that use the new function, you have to add the corresponding dependencies in vcpkg.json file:
````
{
  "name": "vcpkg-cmake",
  "host": true
},
{
  "name": "vcpkg-cmake-config",
  "host": true
}
````

**Comment by DHowett** [comment]
It was my intent to make as small of a change as possible; however, I'll submit another revision that makes the required changes.

**Comment by DHowett** [comment]
I have added a license. This is the officially-recognized SPDX license ID for the bzip2 software license. This license text is identical to the bzip2 1.0.8 license text, _which does not have a unique SPDX identifier._

**Comment by JonLiu1993** [comment]
Feature `tool ` tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static
- x64-linux

**Comment by DHowett** [comment]
Since `tool` is a default feature, it may be a good idea to test `core` without `tool` (`default-features` `false`). Is that covered already? Thanks!

**Comment by JonLiu1993** [comment]
> Since `tool` is a default feature, it may be a good idea to test `core` without `tool` (`default-features` `false`). Is that covered already? Thanks!

Sorry, I was careless and didn't notice this.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by DHowett** [comment]
Thanks all!

**Pull request merged**